src/detect-engine-sigorder.c

```diff
@@ -2082,49 +1900,36 @@ static int SCSigOrderingTest12(void)
 
     uint32_t sids[2] = {1, 2};
     uint32_t results[2] = {1, 1};
-    result = UTHCheckPacketMatchResults(p, sids, results, 2);
-
-end:
-    if (p != NULL)
-        SCFree(p);
-    if (de_ctx != NULL) {
-        SigCleanSignatures(de_ctx);
-        SigGroupCleanup(de_ctx);
-        DetectEngineCtxFree(de_ctx);
-    }
+    FAIL_IF_NOT(UTHCheckPacketMatchResults(p, sids, results, 2));
+
+    SCFree(p);
```

Comment:
noticed this now. Packets allocated by `UTHBuildPacket` should be freed by `PacketFree`. Can you update the tests to do this in the next PR?

Reply:
I was able to find one occurance only.

Reply:
I get the following when I try to free the memory and use it at the end right after `SCFree(p);`
```
suricata(60098,0x102af4580) malloc: *** error for object 0x140035200: pointer being freed was not allocated
suricata(60098,0x102af4580) malloc: *** set a breakpoint in malloc_error_break to debug
```

If I try to free after line 1903 then I get 
```
FLOW MEMORY IN USE 11748608
FAILED
```

Reply:
I think my statement was incorrect. In other tests I see another free function being used:
```c
UTHFreePackets(&p, 1);
```
E.g. see `ModbusParserTest03`

Reply:
I have tried both